### ISSUE TITLE
ID 261624: [Keyboard Navigation - Simon - Alert rules>Show state history]: Contents present on "State history" dialog are not getting scrolled using keyboard.

### ISSUE THREAD

**Issue by eledobleefe** [grafana/grafana#72829]
### **User Experience:** 
Users who depend on keyboards for navigation will get disoriented and they will not be able to scroll content using keyboard and try accessing any element.
_Note: User credentials should NOT be included in the bug._

### **Repro Steps:**
1. Open #Url: [play.grafana.org](https://play.grafana.org/) Sign in with valid credentials. "Home" page appears. 
2. Navigate to the left navigation bar and to the "Alerting" button, select it. It gets expanded and navigates to the expanded sub menu items. 
3. Select the "Alert Rules" menu items. A new page "Alerting" appears. 
4. Navigate the tab " Alert Rules". Select "Data source: AWS IoT Site wise" Navigate to "Grafana" section. 
5. Expand the "Data source: AWS IoT Site wise" button and navigate all the controls present under it. Expand the sub menu in the expanded rule. 
6. Navigate the fields such as, "Labels", "Go to Dashboard" etc. controls. 
7. Navigate to "Show state history" and select it. New dialog "state history" appears. 
8. Navigate all the controls on the dialog. 
9. Observe and verify the issue

**Actual Result:** 
Issue: Scrolling functionality is not working while navigating the controls present on "State history" dialog using keyboard.
Observation: While navigating the controls present on "State history" dialog, Contents are not scrolling down with the arrow keys/pg up/pg dn. Due to which unable to see/access the remaining controls present on the dialog. However, user can scroll the contents(controls) using mouse.

**Expected Result:**

**Solution:** 
Scrolling functionality should work while navigating the controls present on "State history" dialog. So that User can scroll the contents and access the controls present on the dialog using arrow keys/pg up/pg dn.

### **Severity:**
`SEV3`

### **Area:**
`Modal Component (Saga UI Design System)`

**Comment by grafanabot** [comment]
@eledobleefe please add one or more appropriate labels. Here are some tips:

- if you are making an issue, TODO, or reminder for yourself or your team, please add one label that best describes the product or feature area. Please also add the issue to your project board. :rocket:

- if you are making an issue for any other reason (docs typo, you found a bug, etc), please add at least one label that best describes the product or feature that you are discussing (e.g. `area/alerting`, `datasource/loki`, `type/docs`, `type/bug`, etc). [Our issue triage](https://github.com/grafana/grafana/blob/main/ISSUE_TRIAGE.md#3-categorizing-an-issue) doc also provides additional guidance on labeling. :rocket:

 Thank you! :heart:

**Comment by eledobleefe** [comment]
Hi all,
I was about to pick up this issue, but when trying to replicate it, I realised it works as expected. The user can navigate through interactive items (inputs, buttons..) using just the keyboard, they can also use pagination and scroll down the content.

https://github.com/grafana/grafana/assets/65417731/8a8e67d0-8ce7-4f2b-80cf-a771945e99aa

Is this OK? If so, can we close this issue as fixed?

**Comment by eledobleefe** [comment]
> Hi all, I was about to pick up this issue, but when trying to replicate it, I realised it works as expected. The user can navigate through interactive items (inputs, buttons..) using just the keyboard, they can also use pagination and scroll down the content.
> a11y_100124.mov
> 
> Is this OK? If so, can we close this issue as fixed?

Hi again,
This works as expected on local but not on [play.grafana.org](https://play.grafana.org), so I'll dig into it.

**Comment by eledobleefe** [comment]
From a generic point of view, we have "fix" this issue not by coding but by adding the [`Modal` documentation](https://github.com/grafana/design-system/pull/117) to the [Saga website](https://grafana.com/developers/saga/about/overview) since after discussing this in the _a11y working group_ and the _grafana-frontend-platform_ team, we agree we unknow the structure of the content and, therefore, the wanted behaviour in each modal, so the way of solving this is simply giving guidance to developers.

From a specific perspective, we ping the @grafana/alerting-frontend as the issue is related to 'Alert rules > Show state history'. Would you mind taking a look at this?
